### DISCUSSION TITLE
Use neovim built-in language server protocol.

### DISCUSSION THREAD

**Discussion by ghost** [theniceboy/nvim#59] [Ideas]
There is introduction of [built-in lsp](https://neovim.io/doc/user/lsp.html), you also can visit its repository or using command `:help lsp`.

Compare with other lsp client: [there](https://github.com/neovim/nvim-lspconfig/wiki/Comparison-to-other-LSP-ecosystems-(CoC,-vim-lsp,-etc.))

It fully implements the Unix philosophy:

> do one thing and do it well.

So @theniceboy, what do you think of this lsp project and do you want to use it.

**Comment by theniceboy** [comment]
As you know this config uses `coc.nvim` and it's a lot more than just an LSP. My workflow depends on functionalities that extensions of `coc.nvim` provide.

That being said, Neovim's build in LSP is great and fast. I would totally use it if I just needed an LSP client.

**Reply by ghost** [reply]
I understand your means. If so, Can i create a new branch? that use built-in lsp, many coc features can be replaced with other plugins.

Your neovim features will not decrease. You can experience new branch's features.

Is this OK?

**Reply by theniceboy** [reply]
Sounds great! Thank you!

**Reply by ghost** [reply]
And also, i will modularize your neovim config, and write config file with lua script, but it requirement higher neovim version.
